impala/hiveserver2.py

```diff
@@ -566,26 +568,29 @@ def next(self):
         return self.__next__()
 
     def __next__(self):
-        if not self.has_result_set:
-            raise ProgrammingError(
-                "Trying to fetch results on an operation with no results.")
-        if len(self._buffer) > 0:
-            log.debug('__next__: popping row out of buffer')
-            return self._buffer.pop()
-        elif self._last_operation_active:
-            log.debug('__next__: buffer empty and op is active => fetching '
-                      'more data')
-            self._buffer = self._last_operation.fetch(self.description,
-                                                      self.buffersize,
-                                                      convert_types=self.convert_types)
-            if len(self._buffer) == 0:
-                log.debug('__next__: no more data to fetch')
+        while True:
+            if not self.has_result_set:
+                raise ProgrammingError(
+                    "Trying to fetch results on an operation with no results.")
+            if len(self._buffer) > 0:
+                log.debug('__next__: popping row out of buffer')
+                return self._buffer.pop()
+            elif self._last_operation_active:
+                log.debug('__next__: buffer empty and op is active => fetching '
+                          'more data')
+                self._buffer = self._last_operation.fetch(self.description,
+                                                          self.buffersize,
+                                                          convert_types=self.convert_types)
+                if len(self._buffer) > 0:
+                  log.debug('__next__: popping row out of buffer')
+                  return self._buffer.pop()
+                if not self._buffer.has_more_rows:
```

Comment:
nit: I wonder if this was easier to parse if we named the flag `expect_more_rows`

Reply:
Done